lnwallet/channel.go

```diff
@@ -4934,6 +4934,30 @@ func (lc *LightningChannel) AddHTLC(htlc *lnwire.UpdateAddHTLC,
 	return pd.HtlcIndex, nil
 }
 
+// MayAddOutgoingHtlc validates whether we can add an outgoing htlc to this
+// channel. We don't have a value or circuit for this htlc, because we just
+// want to test that we have slots for a potential htlc so we use a "mock"
+// htlc to validate a potential commitment state with one more outgoing htlc.
+func (lc *LightningChannel) MayAddOutgoingHtlc() error {
+	lc.Lock()
+	defer lc.Unlock()
+
+	// Create a "mock" outgoing htlc
+	pd := lc.htlcAddDescriptor(
+		&lnwire.UpdateAddHTLC{
+			Amount: lc.channelState.LocalChanCfg.MinHTLC,
```

Comment:
Wouldn't `validateCommitmentSanity` take that into account for us? Used `MinHtlc` here so that any failure due to balance restrictions would fail for any htlc (since size >= `MinHtlc`). If we move to not counting dust htlcs, I would expect `validateCommitmentSanity` to account for that?

Reply:
I'm not sure I follow, let me know if my thinking isn't correct:
- We move to not count dust HTLCs towards the maximum number of HTLC's in flight
- We predict adding this `MinHTLC` (which may be dust) to the remote commitment transaction
- `validateCommitmentSanity` does not count this HTLC towards max in flight, when it should because the *actual* HTLC we are adding is greater than dust.

But also, this isn't relevant to this PR, just thought I'd point it out.

Reply:
Ah right of course. Good thing to remember when we add `option_dusty_htlcs`, will have to update this check a bit. I think it's good enough in the meantime, since it addresses a pretty gnarley bug?

Reply:
Yup I agree, and there's not consensus yet on `option_dusty_htlcs`, so this is a great stop-gap measure.  (And we'll still have this issue *with* `option_dusty_htlcs`)